widgetssdk/src/main/java/com/glia/widgets/push/notifications/PushNotificationsSetUpUseCase.kt

```diff
@@ -0,0 +1,13 @@
+package com.glia.widgets.push.notifications
+
+internal interface PushNotificationsSetUpUseCase {
+    operator fun invoke(): Boolean
+}
+
+internal class PushNotificationsSetUpUseCaseImpl(private val firebaseAvailabilityCheckManager: FirebaseAvailabilityCheckManager) :
```

Comment:
I don't think that we have so far used interface/impl pattern on usecases and I don't think we should start now.

Reply:
I used this approach for all the use cases for the engagement refactored layer, because this way we can mock it without any problems. But if you have something against this, I can change this one.